build.assets/tooling/cmd/notarize-apple-binaries/gonConfig.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"debug/macho"
+	"encoding/binary"
+	"flag"
+	"io"
+	"os"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type GonConfig struct {
+	AppleUsername string
+	ApplePassword string
+	BinaryPaths   []string
+}
+
+func NewGonConfig() *GonConfig {
+	gc := &GonConfig{}
+	flag.StringVar(&gc.AppleUsername, "apple-username", "", "Apple Connect username used for notarization")
+	flag.StringVar(&gc.ApplePassword, "apple-password", "", "Apple Connect password used for notarization")
+
+	return gc
+}
+
+func (gc *GonConfig) Check() error {
+	err := gc.validateAppleUsername()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate the apple-username flag")
+	}
+
+	err = gc.validateApplePassword()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate the apple-password flag")
+	}
+
+	err = gc.validateBinaryPaths()
+	if err != nil {
+		return trace.Wrap(err, "failed to validate binary path(s)")
+	}
+
+	return nil
+
+	// It might be worth adding an actual login check here in the future
+}
+
+func (gc *GonConfig) validateAppleUsername() error {
```

Comment:
If you use `kingpin` instead you can just tack a `.Required()` on each flag and you won't need all these validation funcs.

Reply:
Some background: I've generally avoided pulling in a third-party CLI parser for `tooling` in order to avoid downloading and compiling it for scripts that are really only ever going to run once in any given context. As these automations get more complex, that's probably a false economy if it means having to write and maintain our own validation routines, though.

Reply:
Note: We all agreed offline to switch to `kingpin`